### ISSUE TITLE
Error Message - Pyfa Won't Install

### ISSUE THREAD

**Issue by Meridiian** [pyfa-org/Pyfa#1238]
I get the following error box when I try to install the latest version of Pyfa on Windows 10 x64.

OS version: 	Windows-8-6.2.9200
Python: 	2.7.10
wxPython: 	3.0.2.0
SQLAlchemy: 	1.0.5
Logbook: 	1.0.0
pyfa version: 1.29.4 Stable - YC119.5 1.0
pyfa root: F:\Eve Tools\pyfa
save path: C:\Users\****\.pyfa
fs encoding: mbcs

EXCEPTION: (sqlite3.OperationalError) table damagePatterns has 7 columns but 9 values were supplied [SQL: 'INSERT INTO damagePatterns VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)'] [parameters: (1, 'Uniform', 25, 25, 25, 25, None, None, None)]

  File "C:\python-2.7.10\lib\site-packages\cx_Freeze\initscripts\Console.py", line 27, in <module>
  File "F:\Eve Tools\pyfa\library.zip\pyfa__main__.py", line 373, in <module>
  File "F:\Eve Tools\pyfa\library.zip\service\prefetch.py", line 40, in <module>
  File "F:\Eve Tools\pyfa\library.zip\eos\db\migration.py", line 41, in update
    func(saveddata_engine)
  File "F:\Eve Tools\pyfa\library.zip\eos\db\migrations\upgrade6.py", line 11, in upgrade
    (1, "Uniform", 25, 25, 25, 25, None, None, None))
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\engine\base.py", line 1989, in execute
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\engine\base.py", line 906, in execute
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\engine\base.py", line 1054, in _execute_text
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\engine\base.py", line 1146, in _execute_context
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\engine\base.py", line 1341, in _handle_dbapi_exception
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\util\compat.py", line 199, in raise_from_cause
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\engine\base.py", line 1139, in _execute_context
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\engine\default.py", line 450, in do_execute

**Comment by blitzmann** [comment]
@Meridiian Sorry for the late reply. Went on vacation and then things got hectic afterwards! 

This was a known issue but thought it was fixed with 1.29.3. Can you send me your `saveddata.db` file to sable.blitzmann@gmail.com? It's located in `%userprofile%\.pyfa\`. I would like to first fix the issue and send it back to you, as well as test it against current code and try to figure out why it's not working.

Or, if you don't have any critical fits and prefer to just delete it, you may do that as well (I would still like a copy to test against though =D)

**Comment by blitzmann** [comment]
Email received and replied to. For documentation:

Problem was due to me overlooking a critical aspect of how the schema changes. 

```
if config.saveDB and os.path.isfile(config.saveDB):
    # If database exists, run migration after init'd database
    pyfalog.debug("Run database migration.")
    db.saveddata_meta.create_all()
    migration.update(db.saveddata_engine)
```

This seemed to be an old database form what I can tell (though it's tough to know for sure). The database did not have the `created` and `modified` fields for `damagePatterns` - which were added in a later migration. When fixing another rareish migration bug involving old databases, I mistakenly assumed `db.saveddata_meta.create_all()` would create the new fields, which it does not (it only creates new tables. Fields need to be manually added).

Fix was to specifically give the insert query the names of the columns, instead of relying on the number of values given.
